doctor/errors.py

```diff
@@ -10,17 +10,22 @@ class TypeSystemError(Exception):
     :param detail: Detail about the error.
     :param cls: The class type that was being instantiated.
     :param code: The error code.
+    :param errobj: A dict containing all validation errors during the request.
```

Comment:
I've never been a huge fan of this name from our internal usage. Maybe it would be better to name it something else for this public version? validation_errors? param_errors? details extra? Not sure...